app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt

```diff
@@ -58,32 +77,70 @@ fun SavedHikesScreen(
       tabList = LIST_TOP_LEVEL_DESTINATIONS,
       selectedItem = Route.SAVED_HIKES,
   ) { paddingValues ->
-    var currentSection by remember { mutableStateOf(SavedHikesScreen.Planned) }
+    var currentSection by remember { mutableStateOf(SavedHikesSection.Planned) }
+    val loading by savedHikesViewModel.loadingSavedHikes.collectAsState()
+    val errorMessageId by savedHikesViewModel.errorMessage.collectAsState()
     val savedHikes by savedHikesViewModel.savedHike.collectAsState()
 
     LaunchedEffect(Unit) { savedHikesViewModel.loadSavedHikes() }
 
     Column(modifier = Modifier.fillMaxSize().padding(paddingValues)) {
-      Column(modifier = Modifier.weight(1f).testTag(TEST_TAG_SAVED_HIKES_SECTION_CONTAINER)) {
-        when (currentSection) {
-          SavedHikesScreen.Planned -> PlannedHikes(savedHikes)
-          SavedHikesScreen.Saved -> SavedHikes(savedHikes)
-        }
-      }
+      Column(
+          modifier =
+              Modifier.weight(1f)
+                  .testTag(SavedHikesScreen.TEST_TAG_SAVED_HIKES_SECTION_CONTAINER)) {
+            when {
+              loading -> LoadingSavedHikes()
+              errorMessageId != null ->
+                  ErrorDisplay(errorMessageId!!) { savedHikesViewModel.loadSavedHikes() }
+              currentSection == SavedHikesSection.Planned -> PlannedHikes(savedHikes)
+              currentSection == SavedHikesSection.Saved -> SavedHikes(savedHikes)
+            }
+          }
 
       // Navigation items between nearby hikes, planned hikes, and saved hikes
       SavedHikesBottomMenu(currentSection) { currentSection = it }
     }
   }
 }
 
+@Composable
+private fun LoadingSavedHikes() {
+  Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+    CircularProgressIndicator(
+        modifier = Modifier.testTag(SavedHikesScreen.TEST_TAG_SAVED_HIKES_LOADING_ANIMATION))
+  }
+}
```

Comment:
Suggestion: We might want to make this a global component. It would probably be useful in other places